shiny/session/_session.py

```diff
@@ -1447,6 +1447,12 @@ async def output_obs():
                     }
                 )
 
+                # Needed so that Shiny knows to flush the outputs. Even if no
+                # outputs/errors are queued, it's necessary to flush so that the
+                # client knows that progress is over.
+                if isinstance(session, AppSession):
+                    await session._flush()
```

Comment:
NOTE: this is a bit different from how we handle this in R ([via `requestFlush()`](https://github.com/rstudio/shiny/blob/ecb591f/R/shiny.R#L1192-L1195)), but we [currently don't have a `request_flush()` in Python](https://github.com/posit-dev/py-shiny/blob/4c28e91dacb3448509943168f35cc6adb8835385/shiny/_app.py#L391-L395)